src/main/java/bio/terra/app/controller/Oauth2ApiController.java

```diff
@@ -0,0 +1,123 @@
+package bio.terra.app.controller;
+
+import bio.terra.app.configuration.OpenIDConnectConfiguration;
+import bio.terra.common.exception.BadRequestException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.api.client.util.Charsets;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.apache.http.message.BasicNameValuePair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.client.RestTemplate;
+
+@Controller
+public class Oauth2ApiController {
+  static final String AUTHORIZE_ENDPOINT = "/oauth2/authorize";
+  static final String TOKEN_REFRESH_ENDPOINT = "/oauth2/token";
+  private static final String SCOPE_PARAM = "scope";
+  private static final String CLIENT_SECRET_PARAM = "client_secret";
+  private final RestTemplate restTemplate;
+  private final OpenIDConnectConfiguration openIDConnectConfiguration;
+
+  @Autowired
+  public Oauth2ApiController(OpenIDConnectConfiguration openIDConnectConfiguration) {
+    this.openIDConnectConfiguration = openIDConnectConfiguration;
+    restTemplate = new RestTemplate(new HttpComponentsClientHttpRequestFactory());
+  }
+
+  @RequestMapping(value = AUTHORIZE_ENDPOINT)
+  public String oauthAuthorize(HttpServletRequest request) {
+    return "redirect:"
+        + openIDConnectConfiguration.getAuthorizationEndpoint()
+        + "?"
+        + decorateAuthRedirectQueryParameters(request.getQueryString());
+  }
+
+  // Modify the query parameters that were passed to be compatible with configured Oauth flow
+  private String decorateAuthRedirectQueryParameters(String initialParams) {
+    List<NameValuePair> parameters = URLEncodedUtils.parse(initialParams, Charsets.UTF_8);
+
+    parameters.addAll(
+        URLEncodedUtils.parse(openIDConnectConfiguration.getExtraAuthParams(), Charsets.UTF_8));
```

Comment:
Just curious if there's a need today for extra auth params in the `/authorize` request? I don't think I saw any (except for the scope added below). It's probably good to have this support if needed though..

Reply:
Yeah, I needed it for adding `prompt=login` (in addition to the scope modification)